src/python/pants/option/global_options.py

```diff
@@ -195,10 +195,8 @@ def register_bootstrap_options(cls, register):
              help='Paths to ignore for all filesystem operations performed by pants '
                   '(e.g. BUILD file scanning, glob matching, etc). '
                   'Patterns use the gitignore syntax (https://git-scm.com/docs/gitignore).')
-    register_enum_option(
-      # TODO: allow using the attribute `GlobMatchErrorBehavior.warn` for more safety!
-      register, GlobMatchErrorBehavior, '--glob-expansion-failure', default='warn',
-      advanced=True,
+    GlobMatchErrorBehavior.register_option(
```

Comment:
Ditto on using `cls`.

Reply:
I have changed all of these to normal `register()` calls!